### PR TITLE
Fix issues with incorrect position for hls dvr

### PR DESCRIPTION
**Pull request by pajong**
For hls dvr on html5 provider, the position was not getting a correct value.
Getting relative position from seekable range to handle hls dvr positions.
JW7-1560

### CODE REVIEW

File: src/js/providers/html5.js [hunk 1]
```diff
@@ -168,8 +168,8 @@ define([
                 return;
             }
 
-            _setBuffered(_getBuffer(), _position, _videotag.duration);
             _updateDuration();
+            _setBuffered(_getBuffer(), _position, _duration);
```
Comment (pajong) [inline src/js/providers/html5.js]: Call update duration first so that we get correct duration value for hls dvr.

File: src/js/providers/html5.js [hunk 2]
```diff
@@ -525,6 +528,10 @@ define([
                 return;
             }
 
+            if (seekPos < 0) {
+                seekPos += _videotag.seekable.start(0) + _videotag.seekable.end(0);
```
Comment (robwalch) [inline src/js/providers/html5.js]: I think you should use `_getSeekableStart() + _getSeekableEnd()` here.

File: src/js/providers/html5.js [hunk 3]
```diff
@@ -436,6 +439,18 @@ define([
             }
         }
 
+        function _getSeekableStart() {
+            var index = _videotag.seekable ? _videotag.seekable.length : 0;
+            var start = Infinity;
+            if (index) {
+                index--;
+                start = _videotag.seekable.start(index);
```
Comment (robwalch) [inline src/js/providers/html5.js]: I don't think you need the above two lines. A length of 1 would still run once.

File: src/js/providers/html5.js [hunk 4]
```diff
@@ -436,6 +439,18 @@ define([
...
+                while(index--) {
+                    start = Math.min(start, _videotag.seeakble.start(index));
+                }
+            }
```
Comment (robwalch) [inline src/js/providers/html5.js]: `return start;` buddy

**Pull request merged**